storage/src/backend/registry.rs

```diff
@@ -438,17 +433,16 @@ impl RegistryReader {
                         HeaderValue::from_str(auth_header.as_str()).unwrap(),
                     );
 
-                    // Try to request registry server with `authorization` header again
-                    let resp = self
+                    // Cache authorization header for next request
+                    // Cache the auth earlier, so other threads have bigger chance to reuse the auth.
+                    self.state.cached_auth.set(auth_header.as_str());
+
+                    // Try to request registry server with `authorization` header again,
+                    // It's exactly a resend of the original request!
```

Comment:
It seems `resend` can't be used as noun.

Reply:
fixed